### PR TITLE
updated libcaer to master, use latest version of libcaer 

### PR DESCRIPTION
**Pull request by federicohyo**
This version  seamlessly work with fx2/3, added function to correctly call datastop on exit, correctly use type instead of packet position. 

I couldn't figure out why the boost::thread_interrup is never catched in driver.cpp line 625 or why the destructor of DavisRosDriver is never called.

### CODE REVIEW

File: davis_ros_driver/include/davis_ros_driver/driver.h [hunk 1]
```diff
@@ -48,6 +48,7 @@ class DavisRosDriver {
 public:
   DavisRosDriver(ros::NodeHandle & nh, ros::NodeHandle nh_private);
   ~DavisRosDriver();
+  void DataStop();
```
Comment (eliasm) [inline davis_ros_driver/include/davis_ros_driver/driver.h]: please change to `dataStop()`

File: davis_ros_driver/src/driver.cpp [hunk 2]
```diff
@@ -153,11 +162,11 @@ void DavisRosDriver::caerConnect()
   /*
```
Comment (eliasm) [inline davis_ros_driver/src/driver.cpp]: Please remove completely (lines 162-169)

File: davis_ros_driver/src/driver.cpp [hunk 3]
```diff
@@ -461,8 +470,10 @@ void DavisRosDriver::readout()
           continue; // Skip if nothing there.
         }
 
+	int type = caerEventPacketHeaderGetEventType(packetHeader);
```
Comment (eliasm) [inline davis_ros_driver/src/driver.cpp]: make `const int`

File: libcaer_catkin/CMakeLists.txt [hunk 4]
```diff
@@ -11,9 +11,8 @@ file(MAKE_DIRECTORY ${CATKIN_DEVEL_PREFIX}/include/libcaer)
 
 ExternalProject_Add(libcaer_src
   GIT_REPOSITORY https://github.com/inilabs/libcaer.git
-  GIT_TAG 57466910da84c25231aab57e99f246239722a628
+  GIT_TAG dc2e8fdb88d40f27f80bb2723d439574efc63a0e
```
Comment (eliasm) [inline libcaer_catkin/CMakeLists.txt]: Maybe use version (e.g. `2.1.3`) instead of commit hash.

Comment (guillermogb) [inline libcaer_catkin/CMakeLists.txt]: How to do this?
the rest of changes have been addressed

**Pull request merged**